src/server/scripts/poller.js

```diff
@@ -0,0 +1,239 @@
+import moment from 'moment';
+import raven from 'raven';
+
+import logging from '../logging';
+import { conf } from '../helpers/config';
+import db from '../db';
+import {
+  getGitHubRepoUrl,
+  parseGitHubRepoUrl
+} from '../../common/helpers/github-url';
+import requestGitHub from '../helpers/github';
+import { internalGetSnapcraftYaml } from '../handlers/launchpad';
+import { getDefaultBranch } from '../handlers/github';
+
+
+const logger = logging.getLogger('poller');
+raven.config(conf.get('SENTRY_DSN')).install();
+
+
+// Process all Repository (DB) models synchronously. Check for changes using
+// `checkSnapRepository` and if changed request a LP snap build and mark
+// it as 'updated'.
+export const pollRepositories = (checker, builder) => {
+  logger.info('GitHub Repository Poller ...');
+  var raven_client = raven.Client(conf.get('SENTRY_DSN'));
+
+  checker = checker || checkSnapRepository;
+  builder = builder || buildSnapRepository;
+
+  // XXX: meh! no ES6 import support, great library :-/
+  let AsyncLock = require('async-lock');
+  let pollRepoLock = new AsyncLock();
+
+  let repoDB = db.model('Repository');
+  return repoDB.fetchAll().then(function (results) {
+    logger.info(`Iterating over ${results.length} repositories.`);
+    results.models.forEach((repo) => {
+      pollRepoLock.acquire('PROCESS-REPO-SYNC', async () => {
+        const owner = repo.get('owner');
+        const name = repo.get('name');
+
+        // XXX skip repos already updated recently, create a new datetime
+        // column exclusively for the poller.
+        const last_polled_at = repo.get('polled_at') || repo.get('updated_at');
+
+        if (!repo.get('snapcraft_name')) {
+          logger.info(`${owner}/${name}: NO NAME IN SNAPCRAFT.YAML`);
+          return;
+        }
+
+        if (!repo.get('store_name')) {
+          logger.info(`${owner}/${name}: NO NAME REGISTERED IN THE STORE`);
+          return;
+        }
+
+        const store_name = repo.get('store_name');
+        const snapcraft_name = repo.get('snapcraft_name');
+        if (store_name != snapcraft_name) {
```

Comment:
These is an edge case related to that where store_name and snapcraft_name (from db) may mismatch, but a recent commit to repo updated the name in snapcraft.yaml to a correct one.
This script currently relies on the state of BSI database that is updated (for the sake of metrics) when user interacts with the UI.

So in such case this database may contain a different state from what really is in GH and LP/Store.
In theory it would be safer to request the build even if names in db don't match, because they may actually match in the code (updated snapcraft.yaml and registered name).
The issue with that would be that for repos where names really don't match (both in db AND in the code) it would trigger unnecessary builds that will show up in UI as build but not uploaded.

The other side of the coin is that there may be a case where the names do match in the db, but snapcraft.yaml was changed to a different name, so it doesn't match anymore. In such case we would request a build (because we think names match) and it will build and fail to upload too.